src/msg/simple/Pipe.cc

```diff
@@ -1704,7 +1709,8 @@ void Pipe::reader()
     }
   }
 
- 
+  rcu_unregister_thread();
```

Comment:
my understanding is a list is maintained of all registered readers, but the list is iterated on the slow path (writes; config changes), and there is no overhead added on reads (the IO path).  I'll add a link to the [URCU interface guide](http://liburcu.org/).